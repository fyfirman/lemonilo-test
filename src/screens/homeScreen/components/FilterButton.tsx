import React from 'react';
import { TouchableNativeFeedback } from 'react-native';
import styled from 'styled-components/native';

const Container = styled.View<{ active: boolean }>`
  padding: ${({ theme }) => theme.spacing(1)}px ${({ theme }) => theme.spacing(2)}px;
  background-color: ${({ theme, active }) =>
    (active ? theme.palette.primary.main : theme.palette.background.grey)};
  margin-right: ${({ theme }) => theme.spacing(1)}px;
  border-radius: ${({ theme }) => theme.borderRadius.small}px;
`;

const Label = styled.Text<{ active: boolean }>`
  color: ${({ theme, active }) => (active ? '#fff' : theme.palette.primary.main)};
  font-weight: bold;
`;

interface FilterButtonProps {
  active?: boolean;
  title: string;
  onPress: () => any;
}

const FilterButton = ({ title, onPress, active = false }: FilterButtonProps) => {
  return (
    <Container active={active}>
      <TouchableNativeFeedback onPress={onPress}>
        <Label active={active}>{title}</Label>
      </TouchableNativeFeedback>
    </Container>
  );
};

export default FilterButton;
