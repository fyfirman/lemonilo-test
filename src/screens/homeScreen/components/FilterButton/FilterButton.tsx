import myTheme from '@theme/theme';
import React from 'react';
import styled from 'styled-components/native';

const Container = styled.Pressable<{ active: boolean }>`
  background-color: ${({ theme, active }) =>
    (active ? theme.palette.primary.main : theme.palette.background.grey)};
  margin-right: ${({ theme }) => theme.spacing(1)}px;
  border-radius: ${({ theme }) => theme.borderRadius.small}px;
`;

const Label = styled.Text<{ active: boolean }>`
  padding: ${({ theme }) => theme.spacing(1)}px ${({ theme }) => theme.spacing(2)}px;
  color: ${({ theme, active }) => (active ? '#fff' : theme.palette.primary.main)};
  font-weight: bold;
`;

interface FilterButtonProps {
  active?: boolean;
  title: string;
  onPress: () => any;
}

const FilterButton = ({ title, onPress, active = false }: FilterButtonProps) => {
  return (
    <Container
      active={active}
      onPress={onPress}
      disabled={active}
      android_ripple={{ color: myTheme.palette.primary.light, radius: 50 }}
    >
      <Label active={active}>{title}</Label>
    </Container>
  );
};

export default FilterButton;
