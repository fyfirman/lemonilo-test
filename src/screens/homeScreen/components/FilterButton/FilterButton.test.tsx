import React from 'react';
import { renderWithTheme } from '../../../../__tests__/helpers';
import FilterButton from './FilterButton';
import 'jest-styled-components';

test('should match with snapshot', () => {
  const tree = renderWithTheme(<FilterButton onPress={jest.fn} title="Button" />);

  expect(tree).toMatchSnapshot();
});
