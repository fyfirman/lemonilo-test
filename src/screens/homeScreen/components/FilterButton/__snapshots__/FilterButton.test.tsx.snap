// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  active={false}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#F5F9F8",
        "borderRadius": 4,
        "marginRight": 8,
      },
    ]
  }
>
  <Text
    active={false}
    style={
      Array [
        Object {
          "color": "#19A0E0",
          "fontWeight": "bold",
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;
